Rubberduck.Parsing/VBA/ParseRunnerBase.cs

```diff
@@ -103,21 +103,19 @@ private void ProcessComponentParseResults(QualifiedModuleName module, Task<Compo
                 var result = finishedParseTask.Result;
                 lock (_state)
                 {
-                    lock (module.Component)
-                    {
-                        _state.SetModuleAttributes(module, result.Attributes);
-                        _state.AddParseTree(module, result.ParseTree);
-                        _state.AddParseTree(module, result.AttributesTree, ParsePass.AttributesPass);
-                        _state.AddTokenStream(module, result.Tokens);
-                        _state.SetModuleComments(module, result.Comments);
-                        _state.SetModuleAnnotations(module, result.Annotations);
-                        _state.AddAttributesRewriter(module, result.AttributesRewriter);
+                    token.ThrowIfCancellationRequested();
+                    _state.SetModuleAttributes(module, result.Attributes);  //This has to come first because it creates the module state if not present.
```

Comment:
personal preference: comments on their own line, unless terrifyingly short

Reply:
I can move it up. However, on the line above, I think the scope of the comment is a bit ambiguous as it refers to this specific line and none below.